readthedocs/projects/tasks.py

```diff
@@ -1315,6 +1315,20 @@ def warn(self, msg):
     delete_queryset.delete()
 
 
+@app.task(max_retries=5, default_retry_delay=7 * 60)
+def clean_build_task(version_id):
+    """Clean the files used in the build of the given version."""
+    version = Version.objects.get_object_or_log(pk=version_id)
+    if not version:
+        return
+    del_dirs = [
+        os.path.join(version.project.doc_path, dir_, version.slug)
+        for dir_ in ('checkouts', 'envs', 'conda')
+    ]
+    for del_dir in del_dirs:
+        broadcast(type='build', task=remove_dirs, args=[(del_dir,)])
```

Comment:
This definitely shouldn't be a broadcast. We only do locking per-server, so doing this will definitely break things.